src/Microsoft.ML.Fairlearn/reductions/UtilityParity.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Data.Analysis;
+
+
+namespace Microsoft.ML.Fairlearn.reductions
+{
+    /// <summary>
+    /// Modeled after the original <see href="https://github.com/fairlearn/fairlearn/blob/931963c40c0ba0cdd1a9e51c29adcc509da224a6/fairlearn/reductions/_moments/utility_parity.py#L45">repo</see>
+    /// A generic moment for parity in utilities (or costs) under classification.
+    /// This serves as the base class for <see cref="DemographicParity">Demographic Parity</see>
+    /// can be used as difference-based constraints or ratio-based constraints.
+    /// 
+    /// Constraints compare the group-level mean utility for each group with the
+    /// overall mean utility
+    /// 
+    /// </summary>
+    public class UtilityParity : ClassificationMoment
```

Comment:
Is this class going to be used by the users? If not we should keep them internal. In general, if the user doesn't need to directly interact with it we should keep things internal.